# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/meson/fix-cross-no-exec.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

Do not try to run clearly cross build (e.g. new libc, different libc) even on
the same CPU arch.

--- ./mesonbuild/compilers/mixins/clike.py.vanilla	2024-08-02 11:40:40.673018714 +0200
+++ ./mesonbuild/compilers/mixins/clike.py	2024-08-02 11:40:42.363018719 +0200
@@ -308,7 +308,7 @@
         if pc.returncode != 0:
             raise mesonlib.EnvironmentException(f'Compiler {self.name_string()} cannot compile programs.')
         # Run sanity check
-        if environment.need_exe_wrapper(self.for_machine):
+        if self.is_cross: # environment.need_exe_wrapper(self.for_machine):
             if not environment.has_exe_wrapper():
                 # Can't check if the binaries run so we have to assume they do
                 return
